### PR TITLE
Add a sidebar to create actions

### PR DESCRIPTION
**Pull request by nicholasdalhaug**

### CODE REVIEW

File: frontend/src/components/Action/ActionCreateSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from 'react'
+
+import { PersonDetails, useApiClients } from '@equinor/fusion'
+import { ModalSideSheet } from '@equinor/fusion-components'
+
+import { Action, Participant, Question } from '../../api/models'
+import ActionCreateForm from './ActionCreateForm'
+import { CircularProgress } from '@equinor/eds-core-react'
+interface Props
```
Comment (Christdej) [inline frontend/src/components/Action/ActionCreateSidebar.tsx:9]: newline

File: frontend/src/components/Action/ActionCreateSidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+const ActionCreateSidebar = ({ open, connectedQuestion, possibleAssignees, onActionCreate, onCloseClick }: Props) => {
+    const apiClients = useApiClients()
+    const [personDetailsList, setPersonDetailsList] = useState<PersonDetails[]>([])
+
+    const isLoading = personDetailsList.length !== possibleAssignees.length
+
+    const getAllPersonDetails = (azureUniqueIds: string[]): Promise<PersonDetails[]> => {
+        const manyPrommises: Promise<PersonDetails>[] = azureUniqueIds.map(azureUniwueId => {
+            return apiClients.people.getPersonDetailsAsync(azureUniwueId).then(response => {
```
Comment (Christdej) [inline frontend/src/components/Action/ActionCreateSidebar.tsx:26]: azureUniwueId => azureUniqueId

File: frontend/src/components/Action/QuestionActionsList.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                                <PriorityIndicator priority={action.priority}/>
+                            </Box>
+                            <Box display="flex" alignItems="center">
+                                <Typography>
+                                    {action.title}
+                                </Typography>
+                            </Box>
+                        </Box>
+
+
```
Comment (Christdej) [inline frontend/src/components/Action/QuestionActionsList.tsx:77]: remove empty lines

File: frontend/src/views/Evaluation/EvaluationGQL.ts [hunk 4]
```diff
@@ -111,6 +111,9 @@ export const useEvaluationQuery = (evaluationId: string): EvaluationQueryProps =
                     answers {
                         ...AnswerFields
                     }
+                    evaluation {
+                        ...EvaluationFields
+                    }
```
Comment (Christdej) [inline frontend/src/views/Evaluation/EvaluationGQL.ts:116]: In fragments, move participant array away from evaluation

File: frontend/src/views/Evaluation/Preparation/PreparationView.tsx [hunk 5]
```diff
@@ -27,19 +61,58 @@ const PreparationView = ({evaluation, onNextStepClick, onProgressParticipant}: P
...
+                <Box display="flex" flexDirection="row">
+                    <Box flexGrow={1}>
+                        <Typography variant="h2">{barrierToString(selectedBarrier)}</Typography>
+                    </Box>
+                    <Box mr={2}>
+                        <ProgressionCompleteSwitch
+                            isCheckedInitially={isParticipantCompleted}
+                            disabled={disableAllUserInput}
+                            onCompleteClick={localOnClompleteClick}
+                            onUnCompleteClick={localOnUnCompleteClick}
```
Comment (Christdej) [inline frontend/src/views/Evaluation/Preparation/PreparationView.tsx:73]: Uncomplete

File: frontend/src/views/Evaluation/Workshop/WorkshopView.tsx [hunk 6]
```diff
@@ -19,6 +27,37 @@ const WorkshopView = ({evaluation, onNextStepClick, onProgressParticipant}: Work
...
+    const disableAllUserInput = isEvaluationFinishedHere
+                                || !participantAllowed
+                                || !hasParticipantBeenHere
+
+    const localOnClompleteClick = () => {
+        const nextProgression = getNextProgression(participantProgression)
+        onProgressParticipant(nextProgression)
+    }
+
+    const localOnUnCompleteClick = () => {
```
Comment (Christdej) [inline frontend/src/views/Evaluation/Workshop/WorkshopView.tsx:55]: Uncomplete

File: frontend/src/views/Evaluation/Workshop/WorkshopView.tsx [hunk 7]
```diff
@@ -31,20 +70,59 @@ const WorkshopView = ({evaluation, onNextStepClick, onProgressParticipant}: Work
...
+                    <Box display="flex" flexDirection="row">
+                        <Box flexGrow={1}>
+                            <Typography variant="h2">{barrierToString(selectedBarrier)}</Typography>
+                        </Box>
+                        <Box mr={2}>
+                            <ProgressionCompleteSwitch
+                                isCheckedInitially={isParticipantCompleted}
+                                disabled={disableAllUserInput}
+                                onCompleteClick={localOnClompleteClick}
+                                onUnCompleteClick={localOnUnCompleteClick}
```
Comment (Christdej) [inline frontend/src/views/Evaluation/Workshop/WorkshopView.tsx:82]: Uncomplete

**Review by Christdej — CHANGES_REQUESTED** [review]

**Review by Christdej — APPROVED** [review]

**Review by ReedOnly — APPROVED** [review]

**Pull request merged**